pkg/apis/networking/types.go

```diff
@@ -257,6 +270,50 @@ type IngressSpec struct {
 	// TODO: Add the ability to specify load-balancer IP through claims
 }
 
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// IngressClass represents the class of the Ingress, referenced by the Ingress
+// Spec.
+type IngressClass struct {
+	metav1.TypeMeta
+	metav1.ObjectMeta
+
+	// Spec is the desired state of the IngressClass.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#spec-and-status
+	// +optional
+	Spec IngressClassSpec
+}
+
+// IngressClassSpec provides information about the class of an Ingress.
+type IngressClassSpec struct {
+	// Controller is responsible for handling this class. This should be
+	// specified as a domain-prefixed path, e.g. "acme.io/ingress-controller".
+	// This allows for different "flavors" that are controlled by the same
+	// controller. For example, you may have different Parameters for the same
+	// implementing controller.
```

Comment:
Is there a reference  on allow characters in the string
Is there a limit on how long this can be

Reply:
Added new domain prefix validation (uses RFC 1123 and 3986) as well as a 250 character limit, thanks!